pkg/sql/resolver.go

```diff
@@ -314,6 +314,14 @@ func (p *planner) HasPrivilege(
 		return false, err
 	}
 
+	err = p.CheckGrantOptionsForUser(ctx, desc, []privilege.Kind{priv.Kind}, priv.GrantOption)
```

Comment:
Is this correct?

Reply:
i think instead it should look like:
```
if priv.GrantOption {
  err = p.CheckGrantOptionsForUser(..., true /* isGrant */)
  // rest of the error checking logic
}
```

also, it might need to be done as part of (or more similarly to) hasPrivilegeFunc below.

best way to make sure it's working right is to run the logic tests:

```
make testbaselogic FILES='privilege_builtins'
```
(could also be with bazel)

Reply:
Also, does this need to be wrapped in a version gate to only work in the new version (reverse of this https://github.com/cockroachdb/cockroach/pull/74172#discussion_r773466386)?